### ISSUE TITLE
Response Appears to be of Type HTTP::Headers (not HTTP::Response::JSON)

### ISSUE THREAD

**Issue by jasel-lewis** [skington/lwp-json-tiny#5]
I greatly appreciate this utility, however, I'm having difficulty with what I feel to be a very simple script.

When I execute the following script:
```pl
#!/usr/bin/perl

use strict;
use warnings;

use Data::Dumper;
use LWP::JSON::Tiny;

my $requestUrl = 'http://date.jsontest.com';
my $userAgent = LWP::UserAgent::JSON->new;
my $request = HTTP::Request::JSON->new(GET => $requestUrl);
my $response = $userAgent->request($request);

print Dumper($response->json_content);
```
... I continue to get the error:
`Can't locate object method "json_content" via package "HTTP::Headers" at /usr/local/share/perl/5.28.1/HTTP/Message.pm line 653.`

My configuration:
- `uname -a`: Linux raspberrypi 4.19.97+ #1294 Thu Jan 30 13:10:54 GMT 2020 armv6l GNU/Linux
- `perl -v`: This is perl 5, version 28, subversion 1 (v5.28.1) built for arm-linux-gnueabihf-thread-multi-64int
- `cpanm LWP::JSON::Tiny --sudo`: LWP::JSON::Tiny is up to date. (0.014)

And not that it should matter, but because it's mentioned in the error message:
- `cpanm HTTP::Headers --sudo`: HTTP::Headers is up to date. (6.22)

**Comment by skington** [comment]
date.jsontest.com, when I open it in a web browser, spits out HTML complaining that it's over quota, so I think what's happening is that you're not, in fact, getting back JSON.

The root cause is that [if the response wasn't JSON, your response object won't be reblessed into the HTTP::Response::JSON class](https://github.com/skington/lwp-json-tiny/blob/master/lib/LWP/UserAgent/JSON.pm#L180), so calling json_content will fail. It will fail with a really annoyingly confusing error message, but that's not something I can do much about at the moment.

**Comment by skington** [comment]
BTW, `$response->as_string` is your friend for debugging stuff like this.

**Comment by skington** [comment]
Also, thanks for using this module that I wrote in a fit of pique!

**Comment by jasel-lewis** [comment]
Sam -

Absolutely right, but probably a comedy of timing.  I was able to get JSON in the browser when I was testing.  This is actually a MUCH more simplified script from what I'm really working on (but did not want to publicly provide the URL and API key I'm using for the real site returning JSON).  I chose the jsontest.com one for pure convenience.

Regardless, I will try `as_string` and see what I can get out of it.

Thank you for the timely response!
